### PR TITLE
Fixed some coding violations

### PR DESCRIPTION
**Pull request by siddhpant**
The links/description of each change is listed under the commit.

### CODE REVIEW

**Comment by siddhpant** [comment]
Merged em successfully

File: xwiki-rendering-syntaxes/xwiki-rendering-syntax-wikimodel/src/main/java/org/xwiki/rendering/internal/parser/wikimodel/DefaultXWikiGeneratorListener.java [hunk 1]
```diff
@@ -225,8 +225,8 @@ private Listener popListener()
             genericParameters = new LinkedHashMap<String, String>();
             for (WikiParameter wikiParameter : params.toList()) {
                 String key = wikiParameter.getKey();
-                if (key.equals(DocumentResourceReference.ANCHOR)
-                    || key.equals(DocumentResourceReference.QUERY_STRING))
+                if ((DocumentResourceReference.ANCHOR).equals(key)
```
Comment (tmortagne) [inline xwiki-rendering-syntaxes/xwiki-rendering-syntax-wikimodel/src/main/java/org/xwiki/rendering/internal/parser/wikimodel/DefaultXWikiGeneratorListener.java]: Useless parenthesis.

File: xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/parser/xhtml/wikimodel/XWikiCommentHandler.java [hunk 2]
```diff
@@ -87,17 +87,19 @@ public void onComment(String content, TagStack stack)
...
+        } else if (content.startsWith("startimage:")) {
             handleImageCommentStart(XMLUtils.unescapeXMLComment(content), stack);
-        } else if (!ignoreElements && content.startsWith("stopimage")) {
+        } else if (content.startsWith("stopimage")) {
             handleImageCommentStop(stack);
-        } else if (!ignoreElements && content.startsWith("startmacro")) {
+        } else (content.startsWith("startmacro")) {
             super.onComment(XMLUtils.unescapeXMLComment(content), stack);
-        } else {
+        } }
```
Comment (tmortagne) [inline xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/parser/xhtml/wikimodel/XWikiCommentHandler.java]: This does not follow XWiki format.

File: xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/parser/xhtml/wikimodel/XWikiCommentHandler.java [hunk 3]
```diff
@@ -87,17 +87,19 @@ public void onComment(String content, TagStack stack)
         // If the comment starts with "startwikilink" then we need to gather all XHTML tags inside
         // the A tag, till we get a "stopwikilink" comment.
         // Same for "startimage" and "stopimage".
-        if (!ignoreElements && content.startsWith("startwikilink:")) {
+        if(!ignoreElements) {
+        if (content.startsWith("startwikilink:")) {
```
Comment (tmortagne) [inline xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/parser/xhtml/wikimodel/XWikiCommentHandler.java]: Wrong indentation.

**Comment by siddhpant** [comment]
Done.

File: xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/parser/xhtml/wikimodel/XWikiCommentHandler.java [hunk 4]
```diff
@@ -87,17 +87,20 @@ public void onComment(String content, TagStack stack)
...
+            } else if (content.startsWith("startimage:")) {
+                handleImageCommentStart(XMLUtils.unescapeXMLComment(content), stack);
+            } else if (content.startsWith("stopimage")) {
+                handleImageCommentStop(stack);
+            } else (content.startsWith("startmacro")) {
+                super.onComment(XMLUtils.unescapeXMLComment(content), stack);
+            }
+        }
+        else {
             super.onComment(content, stack);
```
Comment (tmortagne) [inline xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/parser/xhtml/wikimodel/XWikiCommentHandler.java]: The way you changed the id this is not going to be executed in all places it's supposed to.

Comment (siddhpant) [reply xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/parser/xhtml/wikimodel/XWikiCommentHandler.java]: which id?

Comment (siddhpant) [reply xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/parser/xhtml/wikimodel/XWikiCommentHandler.java]: I just made a parent if condition, which it was going to check anyway multiple times.

**Comment by siddhpant** [comment]
I just copied and pasted the original file. Everything should be fixed now.

**Comment by tmortagne** [comment]
Thanks for your contribution !

**Comment by siddhpant** [comment]
:)

File: xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/parser/xhtml/wikimodel/XWikiCommentHandler.java [hunk 5]
```diff
@@ -127,8 +127,8 @@ private void handleLinkCommentStart(String content, TagStack stack)
 
         stack.pushStackParameter(LINK_LISTENER, xwikiListener);
 
-        stack.pushStackParameter(IS_IN_LINK, true);
-        stack.pushStackParameter(IS_FREE_STANDING_LINK, false);
+        stack.pushStackParameter(IS_IN_LINK, Boolean.TRUE);
+        stack.pushStackParameter(IS_FREE_STANDING_LINK, Boolean.FALSE);
```
Comment (vmassol) [inline xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/parser/xhtml/wikimodel/XWikiCommentHandler.java]: I'm curious to understand why this is better? I find it less good than what we had.

File: xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/parser/xhtml/wikimodel/XWikiXHTMLWhitespaceXMLFilter.java [hunk 6]
```diff
@@ -77,9 +77,9 @@ public void setProperty(String name, Object value) throws SAXNotRecognizedExcept
     public void endCDATA() throws SAXException
     {
         if (getContent().length() > 0 && this.containsWikiSyntax) {
-                // Make sure we clean head/trail white spaces
-                trimLeadingWhiteSpaces();
-                trimTrailingWhiteSpaces();
+            // Make sure we clean head/trail white spaces
```
Comment (vmassol) [inline xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/parser/xhtml/wikimodel/XWikiXHTMLWhitespaceXMLFilter.java]: General remark not related to your commit: This comment is useless. It just explains something obvious... It should explain why it's needed to remove white spaces instead.

File: xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/parser/xhtml/wikimodel/XWikiCommentHandler.java (hunk 5 above)
Comment (siddhpant) [reply xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/parser/xhtml/wikimodel/XWikiCommentHandler.java]: http://sonar.xwiki.org/issues/search#issues=02a0f965-09f1-477f-ab62-6362ab587043

File: xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/XWikiSyntaxChainingRenderer.java [hunk 7]
```diff
@@ -229,7 +229,6 @@ public void beginFormat(Format format, Map<String, String> parameters)
         // If the previous format had parameters and the parameters are different from the current ones then close them
         if (this.previousFormatParameters != null) {
             if (parameters.isEmpty()) {
-                // print("(%%)");
                 // this.previousFormatParameters = null;
```
Comment (vmassol) [inline xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/XWikiSyntaxChainingRenderer.java]: Why remove the first comment and not the second one?

Comment (siddhpant) [reply xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/XWikiSyntaxChainingRenderer.java]: Wait

Comment (tmortagne) [reply xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/XWikiSyntaxChainingRenderer.java]: This is weird, I don't remember this modification when I merged the pull request.

Comment (tmortagne) [reply xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/XWikiSyntaxChainingRenderer.java]: I put it back.

Comment (siddhpant) [reply xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/XWikiSyntaxChainingRenderer.java]: It was a Sonar issue.

Comment (siddhpant) [reply xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/XWikiSyntaxChainingRenderer.java]: http://sonar.xwiki.org/issues/search#issues=9a20df47-672f-4fca-9a38-767978c7d7e6

Comment (tmortagne) [reply xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/XWikiSyntaxChainingRenderer.java]: It's not the right fix for it.

**Pull request merged**